python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -1028,6 +1028,39 @@ cdef class Column:
 
         return Column.from_array_interface(ArrayInterfaceWrapper(iface))
 
+    def to_pylist(self):
+        """
+        Convert the Column to a Python list.
+
+        Only supports fixed-width and string columns.
+
+        Returns
+        -------
+        list
+            A list of Python objects representing the column data.
+
+        Raises
+        ------
+        NotImplementedError
+            If the column type is not fixed-width or string.
+        """
+        cdef unique_ptr[scalar] s
+
+        if not plc_is_fixed_width(self.type()) and self.type().id() != type_id.STRING:
```

Comment:
For fixed width non-null columns, we should probably do the copy in one batch via RMM.